### PR TITLE
SAIC-131 to detect obsolete flavor ID

### PR DESCRIPTION
**Pull request by wzbbbb**
When a flavor is updated, the flavor id will
change. If an instance is in this flavor,
it will keep the old flavor id, can not be
matched to existing flavors
If detected, abort the migration with error
message.

	modified:   cloudferrylib/os/actions/check_needed_compute_resources.py

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/207/

**Comment by wzbbbb** [comment]
Need to update it a bit. will commit again

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/209/

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/CloudFerry_PR_checks/220/

**Comment by miarmak** [comment]
How it can be updated?

**Comment by wzbbbb** [comment]
We can not determine the new flavor ID, unless we also track the flavors with flavor name. So, the patch only detects it and gives a clear message.

**Comment by miarmak** [comment]
@wzbbbb No, I mean, how actually flavor can be updated? I have checked Horizon, CLI and API methods and have not found smth like 'update_flavor'. Maybe I missed smth?

**Comment by wzbbbb** [comment]
just change the specifications of the flavor, for example, add more memory, then save it. The flavor will get new ID.

**Comment by miarmak** [comment]
Ok, I've got it) Smth wrong with my OpenStack) Thanks

File: cloudferrylib/os/actions/check_needed_compute_resources.py [hunk 1]
```diff
@@ -50,3 +54,18 @@ def check(name, units, have, needed):
...
+        # when a flavor is updated, the flavor id will change. If an instance
+        # is in this flavor, it will keep the old flavor id, can not be matched
+        # to existing flavors
+        src_compute = self.src_cloud.resources[utl.COMPUTE_RESOURCE]
+        src_flavors = [flavor.id for flavor in src_compute.get_flavor_list()]
+
+        for instance in objs.values():
+            if instance['instance']['flavor_id'] not in src_flavors:
+                LOG.info("The flavor ID of instance %s does not match to any"
+                         " flavor. Pleas resize it.",
```
Comment (miarmak) [inline cloudferrylib/os/actions/check_needed_compute_resources.py]: I guess resizing will not return old flavor id...

Comment (wzbbbb) [inline cloudferrylib/os/actions/check_needed_compute_resources.py]: resizing will make it use the new flavor ID.

**Comment by Wasapl** [comment]
Merged build finished. Test FAILed.   http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/8/

File: cloudferrylib/os/actions/check_needed_compute_resources.py (hunk 1 above)
Comment (miarmak) [inline cloudferrylib/os/actions/check_needed_compute_resources.py]: Ok, I've got it. Thanks. 
Just 3 small nits here:
1. Typo in word 'please'
2. I guess we should use another log level for such case (Critical or Error)
3. Could you change message 'Pleas resize it' to 'Please resize the VM', cause f.e. it was not so obvious for me.

**Comment by roman-verchikov** [comment]
@wzbbbb please update according to @miarmak's comments

**Comment by wzbbbb** [comment]
Will create a new pull request SAIC-131a

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.  http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/22/

**Pull request closed without merging**